kube-runtime/src/wait.rs

```diff
@@ -249,68 +234,44 @@ pub mod conditions {
     #[must_use]
     pub fn is_deployment_completed() -> impl Condition<Deployment> {
         |obj: Option<&Deployment>| {
-            if let Some(depl) = &obj {
-                if let Some(s) = &depl.status {
-                    if let Some(conds) = &s.conditions {
-                        if let Some(dcond) = conds.iter().find(|c| {
-                            c.type_ == "Progressing" && c.reason == Some("NewReplicaSetAvailable".to_string())
-                        }) {
-                            return dcond.status == "True";
-                        }
-                    }
-                }
-            }
-            false
+            let conds = obj?.status.as_ref()?.conditions.as_ref()?;
+            let progressing = conds.iter().find(|c| {
+                c.type_ == "Progressing" && c.reason == Some("NewReplicaSetAvailable".to_string())
+            })?;
+            Some(progressing.status == "True")
         }
     }
 
     /// An await condition for `Service`s of type `LoadBalancer` that returns `true` once the backing load balancer has an external IP or hostname
     #[must_use]
     pub fn is_service_loadbalancer_provisioned() -> impl Condition<Service> {
         |obj: Option<&Service>| {
-            if let Some(svc) = &obj {
-                // ignore services that are not type LoadBalancer (return true immediately)
-                if let Some(spec) = &svc.spec {
-                    if spec.type_ != Some("LoadBalancer".to_string()) {
-                        return true;
-                    }
```

Comment:
hey @detjensrobert doing a follow-up on conditions. wondering if you had a particular reason to return true here rather than false for this condition. feels to me we shouldn't consider a condition "matching" on a service that's not even of the right type.

also if you have opinions on this pr, happy to hear 🙏 

EDIT: sorry about spam, put comment on wrong place with wrong account.

Reply:
My thought was that since non-LoadBalancer services do not need to wait for a cloud LB, they are (almost) immediately ready and do not need any checks. I can see this wanting to fail on the wrong type of service though.